sklearn/utils/validation.py

```diff
@@ -341,9 +341,14 @@ def check_array(array, accept_sparse=None, dtype="numeric", order=None,
     X_converted : object
         The converted and validated X.
     """
+    if isinstance(array, np.memmap) and array.mode == "r" and not copy:
+        raise ValueError('Can not modify inplace an array which is read-only')
+
```

Comment:
this is not the right place to raise that error.
check_array method can be called in both `fit()` and `transform()`.
The returned array in inplace modified in `transform()` but no modification seem to be performed in `fit()`.
Could we make this check at the beginning of each `transform()` method ? WDYT @ogrisel 
This may be in the mother class's method fit_transform() and used calling super()